plugin/manager.go

```diff
@@ -69,17 +69,25 @@ func (p *Manager) KillProvider(providerName string) error {
 func (p *Manager) GetOrCreateProvider(providerName, version string) (CQProvider, error) {
 
 	provider, err := p.GetProvider(providerName, version)
-	if provider != nil {
+	if provider != nil && err != nil {
 		return provider, err
 	}
-
-	p.lock.Lock()
-	version = ""
-	if version == "" {
-		version = "latest"
+	// Create RPC client and initialize CQProvider
+	provider, err = p.createProvider(providerName, version)
+	if err != nil {
+		return nil, err
 	}
```

Comment:
I think it's possible there will be a race condition here. between create a provider and set a provider.